### ISSUE TITLE
Possible to determine whether an e-mail address has an associated account

### ISSUE THREAD

**Issue by charmander** [Weasyl/weasyl#243]
When signing up or attempting to change e-mail addresses, if an address is in use on Weasyl, the request is denied.

**Comment by kfkitsune** [comment]
I'm going to tentatively say that if we are going for complete plausible deniability over the existence of addresses here, the forgotten password path seems to apply here too. Ignoring the fact that usernames are inherently public, from [weasyl/resetpassword.py#L36](https://github.com/Weasyl/weasyl/blob/master/weasyl/resetpassword.py#L36), where ``user`` is the result of a SQL query retrieving the username and email based on the supplied username:

```
# Check the user's email address against the provided e-mail address,
# raising an exception if there is a mismatch
if email != emailer.normalize_address(user.email):
    raise WeasylError("emailInvalid")
```

In a roundabout way, this could be used the same way: given an email address $EMAIL, and a list of candidate user names $NAMES,

```
for user_name in $NAMES:
    attempt resetting password for user_name with email address $EMAIL
```

It's a tad noisy, potentially, but the same could be said for potentially brute-forcing the process to find out which address belongs to a given account.

A better option might be, over in [weasyl/controllers/user.py#L282](https://github.com/Weasyl/weasyl/blob/master/weasyl/controllers/user.py#L282), saying "**Success!** Provided the supplied information matched our records, information on how to reset your password has been sent to your email address." -- Optionally, provide the email address the user provided on the reset page (" ... to your email address, " + form.email), so they have immediate feedback if they incorrectly entered their email address, since we are essentially making the process a dark grey box (they know they entered their username correctly, but their email could potentially be entered incorrectly (and we don't tell them that)).

Just my two cents, here.

**Comment by kfkitsune** [comment]
>  change e-mail addresses

I've just been thinking about this... and I am not entirely sure if there's a way to not leak the existence of an email here with the code as it is currently written (when viewing this issue as a completely isolated topic). If we get #225 reviewed/merged, then that changes things and we can have plausible deniability since we wouldn't be performing an immediate change of the email address if it was to an otherwise valid email.

As it is currently, if the email is changed, the user will know if the change happened or not because the email address would update in the settings, or if it failed because the email address existed (from the WeasylError throwing). With #225 in the mix, we could claim that we sent the verification email to the provided email, but just not send it out as _we_ know the email address is in use.

**Comment by charmander** [comment]
Correct; this depends on #220.

**Comment by kfkitsune** [comment]
Fixed in merged PR #276.
